src/nwm_routing/src/nwm_routing/__main__.py

```diff
@@ -1197,6 +1152,14 @@ def main_v03(argv):
             showtiming,
             verbose,
             debuglevel,
+            run_results,
+            data_assimilation_parameters.get("wrf_hydro_lastobs_file",False),
+            q0,
+            gages,
+            data_assimilation_parameters.get('lastobs_output_folder',False),
+            run_set_iterator,
+            main_start_time,
+            lastobs_output_title,
```

Comment:
can you calculate this at a lower level? Like in the function that actually writes the output?

Reply:
the function in nhd_io writes the output. It is called from output.py. The write-out requires all of these different variables in order to run. I was originally asked to move it to output in order to maintain the writing out of lastobs files between v02-3 and keep it inside of the async code. Let me know how you would like to proceed. Line 1090 of nhd_io contains the outputting functionality.